js-pkg/client/src/provider.ts

```diff
@@ -398,6 +485,7 @@ export class YSweetProvider {
   private websocketClose(event: CloseEvent) {
     this.emit(EVENT_CONNECTION_CLOSE, event)
     this.setStatus(STATUS_ERROR)
+    this.clearHeartbeat()
```

Comment:
I think we want to clear the connection timeout here as well:
```suggestion
    this.clearHeartbeat()
    this.clearConnectionTimeout()
```